### PR TITLE
Profile: add the update profile option in manage profile

### PR DESCRIPTION
**Pull request by AmishaBisht**
## Summary
 Target issue is #3146
 added the update profile option in the manage profile node, now the user can update their name and type in profile

### CODE REVIEW

File: lib/glific/flows/action.ex [hunk 1]
```diff
@@ -40,8 +40,9 @@ defmodule Glific.Flows.Action do
 
   @contact_profile %{
     "Switch Profile" => :switch_profile,
-    "Create Profile" => :create_profile
-  }
+    "Create Profile" => :create_profile,
+    "update profile" => :update_profile
```
Comment (mdshamoon) [inline lib/glific/flows/action.ex]: Please check the pattern for the cases in naming and update

File: lib/glific/profiles.ex [hunk 2]
```diff
@@ -231,10 +231,46 @@ defmodule Glific.Profiles do
...
   end
 
+  def handle_flow_action(:update_profile, context, action) do
+    attrs = %{
+      name: ContactField.parse_contact_field_value(context, action.value["name"]),
+      type: ContactField.parse_contact_field_value(context, action.value["type"])
+    }
+
+    case Repo.fetch_by(Profile, %{contact_id: context.contact.id}) do
+      {:ok, update_profile} ->
```
Comment (mdshamoon) [inline lib/glific/profiles.ex]: What happens if a contact has multiple profiles

Comment (AmishaBisht) [reply lib/glific/profiles.ex]: It would update for the current profile

Comment (mdshamoon) [reply lib/glific/profiles.ex]: I am not sure how is it fetching the current profile of the contact. This code seems to fetch any profile of the contact.

File: lib/glific/flows/action.ex [hunk 3]
```diff
@@ -40,7 +40,8 @@ defmodule Glific.Flows.Action do
 
   @contact_profile %{
     "Switch Profile" => :switch_profile,
-    "Create Profile" => :create_profile
+    "Create Profile" => :create_profile,
+    "Update profile" => :update_profile
```
Comment (mdshamoon) [inline lib/glific/flows/action.ex:45]:
```suggestion
    "Update Profile" => :update_profile
```

Keep it consistent

**Review by mdshamoon — CHANGES_REQUESTED** [review]

**Review by mdshamoon — COMMENTED** [review]
Add test cases for cases not covered.

**Comment by AkhileshNegi** [comment]
Closing stale PR

**Pull request closed without merging**